packages/react-dom/src/events/plugins/__tests__/ModernBeforeInputEventPlugin-test.js

```diff
@@ -825,4 +825,36 @@ describe('BeforeInputEventPlugin', () => {
   it('should extract onBeforeInput when simulating in env with only CompositionEvent on contenteditable', () => {
     testContentEditableComponent(environments[3], scenarios);
   });
+
+  describe('supporting the native beforeinput event', () => {
+    beforeEach(() => {
+      jest.resetModules();
+      InputEvent.prototype.getTargetRanges = function() {};
+      React = require('react');
+      ReactDOM = require('react-dom');
+    });
+
+    afterEach(() => {
+      delete InputEvent.prototype.getTargetRanges;
+    });
+
+    function dispatchBeforeInputEvent(target, data) {
+      const event = new InputEvent('beforeinput', {bubbles: true, data});
+      target.dispatchEvent(event);
+    }
+
+    it('should extract onBeforeInput on an input', () => {
+      const onBeforeInput = jest.fn();
```

Comment:
Might be nice to actually test that the data gets passed through to the event handler; perhaps something like this:

```suggestion
      const onBeforeInput = jest.fn().mockImplementationOnce((e) => {
        expect(e.data).toBe('A');
      });
```